fe/fe-core/src/main/java/org/apache/doris/common/UUIDv7Generator.java

```diff
@@ -0,0 +1,124 @@
+// 3. Global uniqueness - Combines timestamp with random data to ensure uniqueness across
+//    distributed systems without coordination
+// 4. Database friendly - The time-ordered nature makes it more efficient for database indexing
+//    and storage compared to purely random UUIDs (like v4)
+// 5. Future-proof - Follows the latest UUID standard with improvements over older versions
+
+// Note: Our implementation differs slightly from the standard UUIDv7 specification by
+// using a counter instead of random bits in the "rand_a" field to further enhance
+// uniqueness when generating multiple UUIDs in rapid succession.
+
+// 0                   1                   2                   3
+// 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |                           unix_ts_ms                          |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |          unix_ts_ms           |  ver  |       counter         |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |var|                        rand_b                             |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+// |                            rand_b                             |
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.LockSupport;
+
+public class UUIDv7Generator {
+    private static final UUIDv7Generator INSTANCE = new UUIDv7Generator();
+    private static final Random RANDOM = ThreadLocalRandom.current();
+    private static final AtomicInteger COUNTER = new AtomicInteger(0);
+    private static final long VERSION = 7L << 12;
+    private static final long VARIANT = 2L << 62;
+    private static final long MAX_SEQUENCE = 0xFFF;
+    private static final int SPIN_THRESHOLD = 10;
+    private static final long PARK_NANOS = 100L; // 100 nanoseconds
+
+    private final AtomicLong timestampAndSequence;
+
+    private UUIDv7Generator() {
+        // Initialize with current timestamp and 0 sequence
+        long initialTimestamp = System.currentTimeMillis();
+        timestampAndSequence = new AtomicLong((initialTimestamp << 12));
+    }
+
+    public static UUIDv7Generator getInstance() {
+        return INSTANCE;
+    }
+
+    public UUID nextUUID() {
+        int spinCount = 0;
+        while (true) {
+            long currentTimestamp = System.currentTimeMillis();
+            long current = timestampAndSequence.get();
+            long timestamp = current >>> 12;
+            long sequence = current & MAX_SEQUENCE;
+
+            if (currentTimestamp < timestamp || (currentTimestamp == timestamp && sequence >= MAX_SEQUENCE)) {
```

Comment:
The algorithm in Java and C++ parts seems different. Should it be syncronized? If not, could you please add a comment documenting these differences (mainly *why*)